### DISCUSSION TITLE
Unit test for Binding Reducer

### DISCUSSION THREAD

**Discussion by CharanPera** [pointfreeco/swift-composable-architecture#3134] [Q&A] [answered]
Hi, How can i write Unit test for the below reducer

`case binding(BindingAction<State>)`

```
case .binding(\.pickerSelection):
  return .run { [value = state.pickerSelection] send in
        await send(.pickerChanged(value))
   }
```

**Comment by mbrandonw** [comment, marked as answer]
Hi @CharanPera, before telling you how to write a test for that I would want to recommend that perhaps you do not need the `pickerChanged` action. You are sending an action to tell the reducer that something changed, but the `.binding(\.pickerSelection)` action _is_ exactly that already. Whatever logic you want to do in `.pickerChanged` can just be done in `.binding(\.pickerSelection)`.

However, having said that, you would test this code like so:

```swift
// Assuming a 'store = TestStore(…)' has been set up

await store.send(.binding(\.pickerSelection, 1)) {
  // Assert on anything that changed
}
await store.receive(\.pickerChanged) {
  // Assert on anything that changed
}
```
